StRoot/StFwdUtils/StFwdAnalysisMaker.cxx

```diff
@@ -49,30 +55,101 @@
 #include "StarClassLibrary/SystemOfUnits.h"
 
 
+#include "TROOT.h"
+#include "TLorentzVector.h"
 #include "StEvent/StFwdTrack.h"
+#include "StFcsDbMaker/StFcsDb.h"
 
 //________________________________________________________________________
 StFwdAnalysisMaker::StFwdAnalysisMaker() : StMaker("fwdAna"){};
-int StFwdAnalysisMaker::Finish() { return kStOk; }
+int StFwdAnalysisMaker::Finish() { 
+    
+    auto prevDir = gDirectory;
+        
+    // output file name
+    string name = "StFwdAnalysisMaker.root";
+    TFile *fOutput = new TFile(name.c_str(), "RECREATE");
+    fOutput->cd();
+    for (auto nh : mHists) {
+        nh.second->SetDirectory(gDirectory);
+        nh.second->Write();
+    }
+
+    // restore previous directory
+    gDirectory = prevDir;
+
+    LOG_INFO << "Writing StFwdAnalysisMaker output" << endm;
```

Comment:
My suggestion depends on what you want from the log statement....
- If you want to say your code has made it to the writing stage, you generally put this before the `Write()` call.
- If you want to say your code has completed the writing stage, you generally output something like "Done writing StFwdAnalysisMaker output", or some variation on that.

Reply:
Updated in [030a2f5](https://github.com/star-bnl/star-sw/pull/636/commits/030a2f51a0cb1aa2368fac8ec6070243646aeda0)